inventory/agents_test.go

```diff
@@ -0,0 +1,429 @@
+package inventory
+
+import (
+	"context"
+	"testing"
+
+	"github.com/percona/pmm/api/inventory/json/client"
+	"github.com/percona/pmm/api/inventory/json/client/agents"
+	"github.com/percona/pmm/api/inventory/json/client/services"
+	"github.com/stretchr/testify/require"
+
+	_ "github.com/Percona-Lab/pmm-api-tests" // init default client
+)
+
+func TestAgents(t *testing.T) {
+	t.Run("List", func(t *testing.T) {
+		t.Parallel()
+
+		nodeID := addRemoteNode(t, "Remote node for agents list")
+		defer removeNodes(t, nodeID)
+
+		serviceID := addMySQLService(t, services.AddMySQLServiceBody{
+			NodeID:      "pmm-server",
+			Address:     "localhost",
+			Port:        3306,
+			ServiceName: "MySQL Service for agent",
+		})
+		defer removeServices(t, serviceID)
+
+		mySqldExporter := addMySqldExporter(t, agents.AddMySqldExporterBody{
+			ServiceID:    serviceID,
+			Username:     "username",
+			Password:     "password",
+			RunsOnNodeID: "pmm-server",
+		})
+		mySqldExporterID := mySqldExporter.MysqldExporter.AgentID
+		defer removeAgents(t, mySqldExporterID)
+
+		pmmAgent := addPMMAgent(t, nodeID)
+		pmmAgentID := pmmAgent.Payload.PMMAgent.AgentID
+		defer removeAgents(t, pmmAgentID)
+
+		res, err := client.Default.Agents.ListAgents(&agents.ListAgentsParams{Context: context.TODO()})
+		require.NoError(t, err)
+		require.NotNil(t, res)
+		require.NotZerof(t, len(res.Payload.MysqldExporter), "There should be at least one service")
+
+		require.Conditionf(t, func() (success bool) {
+			for _, v := range res.Payload.MysqldExporter {
+				if v.AgentID == mySqldExporterID {
+					return true
+				}
+			}
+			return false
+		}, "There should be MySQL agent with id `%s`", mySqldExporterID)
+		require.Conditionf(t, func() (success bool) {
+			for _, v := range res.Payload.PMMAgent {
+				if v.AgentID == pmmAgentID {
+					return true
+				}
+			}
+			return false
+		}, "There should be PMM-agent with id `%s`", pmmAgentID)
+	})
+
+	t.Run("FilterList", func(t *testing.T) {
+		t.Parallel()
+
+		nodeID := addRemoteNode(t, "Remote node for agents filters")
+		defer removeNodes(t, nodeID)
+
+		serviceID := addMySQLService(t, services.AddMySQLServiceBody{
+			NodeID:      "pmm-server",
+			Address:     "localhost",
+			Port:        3306,
+			ServiceName: "MySQL Service for filter test",
+		})
+		defer removeServices(t, serviceID)
+
+		mySqldExporter := addMySqldExporter(t, agents.AddMySqldExporterBody{
+			ServiceID:    serviceID,
+			Username:     "username",
+			Password:     "password",
+			RunsOnNodeID: "pmm-server",
+		})
+		mySqldExporterID := mySqldExporter.MysqldExporter.AgentID
+		defer removeAgents(t, mySqldExporterID)
+
+		pmmAgent := addPMMAgent(t, nodeID)
+		pmmAgentID := pmmAgent.Payload.PMMAgent.AgentID
+		defer removeAgents(t, pmmAgentID)
+
+		// Filter by runs on node ID.
+		res, err := client.Default.Agents.ListAgents(&agents.ListAgentsParams{
+			Body:    agents.ListAgentsBody{RunsOnNodeID: "pmm-server"},
```

Comment:
`pmm-server` is a special case we likely to remove in the release. Do not count on it to be present.